sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/GeneralScalarExpression.java

```diff
@@ -190,6 +113,19 @@ public GeneralScalarExpression(String name, Expression[] children) {
   public String name() { return name; }
   public Expression[] children() { return children; }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    GeneralScalarExpression that = (GeneralScalarExpression) o;
+    return Objects.equals(name, that.name) && Arrays.equals(children, that.children);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(name, children);
```

Comment:
@beliefer , should it be `Arrays.hashCode(children)`?

Reply:
`Arrays.hashCode(children)` only get part hash code from children. `Objects.hash(name, children)` considers the name too.

Reply:
Sorry, I meant `return Objects.hash(name, Arrays.hashCode(children));`. Otherwise hash of array `children` is just a reference hash, i.e. it does not consider elements' hashes.

Reply:
I got it. Thank you for the comment.

Reply:
I will fix it. Thanks.